### DISCUSSION TITLE
Plugin now Freemium ?

### DISCUSSION THREAD

**Discussion by Julienraptor01** [dsoftwareinc/ghactions-manager#159] [Q&A] [answered]
i just saw in my IDE that the plugin has gone from free to freemium, what does it change, what are the feature that where free and become paid / what are the new paid features ?

**Comment by cunla** [comment, marked as answer]
Hi, thanks for bringing this up. I updated the readme with the details but better include them here as well.

The gist:
Features from past versions remain free. There is currently only one new paid feature (downloading workflow-run artifacts), but there are several features I am working on which will be paid.

> There is a 50% discount for users who purchase the plugin before June 30, 2024.
> Use this discount code: DWKJY-CEHXB-6FMZ3-PZ9B5-PP8WK.

Full list:

# Features

### Main use-case

- View the latest workflow runs and their statuses.
  - Automatically refreshes every 30 seconds.
  - Can filter by: workflow-type, user who initiated, branch and event.
  - Can filter by the current-branch, i.e., update workflow-runs when branch changes, and current user.
- View jobs of a workflow-run and their statuses.
- View logs of a job, separated by each step result.
- (Paid feature) View and download the workflow run's artifacts.

### Additional use-cases

- Trigger a workflow-run.
- Rerun a workflow run or a job.
- Open the workflow file of the workflow-run.
- Configuring settings
  - Use a customized token instead of GitHub settings
  - Frequency of auto-refresh.
  - Customize repositories that should be presented/hidden.

### Editing workflow files

- Highlight outdated actions as warnings
  - Generate quickfix: update them to the latest version.
- Highlight unknown actions
- Navigate to action repository for remote actions
- Navigate to action file for local actio

**Reply by Julienraptor01** [reply]
Can i ask what the reason behind the force push of the commit history and the deletion of all the old free code is ?
Couldn't it have been kept on an archive branch ?

**Reply by cunla** [reply]
Looks like you managed to fork before the removal, so you are good :)
Keep in mind the license for the code is GPLv3. 

There are 2 main reasons:
- I don't want to deal with people creating copies of the plugin, and while the code was available, and to some extent still available, it was and still is copyrighted.
- Beforehand I published releases both in JetBrains and in GitHub. I realized there is no way to publish releases in GitHub without having the code, and it would mean the last release in the repo is 1.22.0 - which I didn't want to confuse people with. (I know personally 2 people who install the plugin by downloading it from GitHub).

**Reply by Julienraptor01** [reply]
> Looks like you managed to fork before the removal, so you are good :)

I actually made my fork after the removal, but github is smart and still keep the commits after force push, and you can find the old commit trees with the API
> Keep in mind the license for the code is GPLv3.

Indeed and it's very nice

> There are 2 main reasons:
> - I don't want to deal with people creating copies of the plugin, and while the code was available, and to some extent still available, it was and still is copyrighted.

Yeah the old code is unde the GPLv3 license which allow people to fork, modify and redistribute as long as they follow it
> - Beforehand I published releases both in JetBrains and in GitHub. I realized there is no way to publish releases in GitHub without having the code, and it would mean the last release in the repo is 1.22.0 - which I didn't want to confuse people with. (I know personally 2 people who install the plugin by downloading it from GitHub).

You can actually publish releases all based on the same init commit, without having any code change (i just tested on a blank repo and i was able to create 3 different releases one after the other without pushing anything to the repo, so that the only file present was the default readme) 

Anyway, thanks for the answer and i wish you a great continuation

**Comment by mpuglin** [comment]
Very vague licensing terms, imo.  Should clarify...  

If Plugin is a paid Plugin, you must ensure that the maximum number of Plugin Users does not exceed the number of Plugin Users for which you have purchased Plugin.

You are authorized to use Plugin in accordance with its documentation provided by Developer and for the period of time specified by Developer.

**Reply by cunla** [reply]
Hi @mpuglin,

Thanks for reaching out with this.
Where did you find the licensing terms? I suspect this is a generic text provided by JetBrains and not something I can control.

The plugin licensing terms are provided in [the JetBrains plugin page](https://plugins.jetbrains.com/plugin/19347-github-actions-manager/pricing)
![image](https://github.com/user-attachments/assets/e7781006-4440-436b-a43e-2a05f6cc813c)

**Reply by mpuglin** [reply]
Hi, thanks for response .

Found this in your license file on GitHub

**Reply by cunla** [reply]
I see, I'll remove the LICENSE file from repository.
Thanks for bringing this up.
